game-core/src/main/java/games/strategy/util/CollectionUtils.java

```diff
@@ -98,6 +99,9 @@ private CollectionUtils() {}
    * Note that (a,a,b) (a,b,b) are equal.
    */
   public static <T> boolean equals(final Collection<T> c1, final Collection<T> c2) {
-    return Objects.equals(c1, c2) || (c1.size() == c2.size() && c2.containsAll(c1) && c1.containsAll(c2));
+    checkNotNull(c1);
+    checkNotNull(c2);
+
+    return Iterables.elementsEqual(c1, c2) || (c1.size() == c2.size() && c2.containsAll(c1) && c1.containsAll(c2));
```

Comment:
I wonder if we really should just throw all elements in a guava MultiSet and compare the 2 sets for equality.
I'd guess the performance will be better for larger collections, not sure if collections are using this method that are actually large enough though.

But maybe it's just that I really don't like this method: I can't come up with a reason that justifies having to create 2 collections that might contain the same elements with potential duplicates but in potentially different orders 🤷‍♂️

Reply:
I feel your pain. :smile:  Every time I come across this method, I hope it's no longer used so I can just remove it.  I don't really understand the game logic need for it (although I'm sure it's legitimate).  Before changing the implementation, I'd want to do a deeper analysis to better understand the arguments it's actually called with.  But I think that's outside the scope of this PR. 

If anything, I think it needs to be renamed because `equals()` alone is completely misleading.  I ask myself each time, "Why didn't they just call `Object#equals()`?" before having to digest the implementation and realizing it's not quite the same as `Object#equals()`.

I went ahead and renamed it to `haveEqualSizeAndEquivalentElements()` in c3e98781c to be explicit about its purpose (and to hopefully better draw attention to its smell).  Let me know if you disagree.